feeds/scheduler/feed_group.go

```diff
@@ -0,0 +1,115 @@
+package scheduler
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/ossf/package-feeds/feeds"
+	"github.com/ossf/package-feeds/publisher"
+)
+
+type FeedGroup struct {
+	feeds     []feeds.ScheduledFeed
+	publisher publisher.Publisher
+	lastPoll  time.Time
+}
+
+func NewFeedGroup(scheduledFeeds []feeds.ScheduledFeed,
+	pub publisher.Publisher, initialCutoff time.Duration) *FeedGroup {
+	return &FeedGroup{
+		feeds:     scheduledFeeds,
+		publisher: pub,
+		lastPoll:  time.Now().Add(-initialCutoff),
+	}
+}
+
+func (fg *FeedGroup) Run() {
+	_, err := fg.PollAndPublish()
+	if err != nil {
+		log.Error(err)
+	}
+}
+
+func (fg *FeedGroup) PollAndPublish() (int, error) {
+	pkgs, errs := fg.poll()
+	if len(errs) > 0 {
+		return 0, errs[0]
+	} else if len(pkgs) == 0 {
+		return 0, nil
+	}
+
+	log.WithField("num_packages", len(pkgs)).Printf("Publishing packages...")
+	numPublished, err := fg.publishPackages(pkgs)
+	if err != nil {
+		log.Errorf("Failed to publish %v packages due to err: %v", len(pkgs)-numPublished, err)
+		return numPublished, err
+	}
+	log.WithField("num_packages", numPublished).Printf("Successfully published packages")
+	return numPublished, nil
+}
+
+// Poll fetches the latest packages from each registered feed.
+func (fg *FeedGroup) poll() ([]*feeds.Package, []error) {
+	results := make(chan pollResult)
```

Comment:
I think we may want to buffer the channel `make(chan pollResult, len(fg.feeds))`